src/filterBuilder.js

```diff
@@ -58,8 +58,13 @@ export class FilterBuilder {
   }
 
   async buildFilter() {
-    const adapter = getPermissionsAdapter()
-    this.userData = await adapter.fetchUserPermissions()
+    if (this.bypassPermissions) {
```

Comment:
@adrian-musora - I think we should not pull the users permissions unless they are needed for built filter. Makes testing a lot easier as well. Let me know if you see any issues with this approach.